src/averageAges.js

```diff
@@ -15,11 +15,19 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  let peopleArr = [...people];
+
+  (century) && (peopleArr = people.filter(
+    person => Math.ceil(person.died / 100) === century)
+  );
+
+  const onlyMen = peopleArr.filter(person => person.sex === 'm')
+    .map(person => person.died - person.born);
```

Comment:
```suggestion
  const onlyMen = people.filter(person => century
    ? возвращаем всех у кого sex === 'm' и нужное значение century
    : просто возвращаем всех m
  ).map(person => person.died - person.born);
```
Можно сократить вот так